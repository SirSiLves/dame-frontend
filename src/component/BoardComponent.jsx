import React, {Component} from 'react'
import FieldLineComponent from "./FieldLineComponent";
import GameDataService from "../service/GameDataService";
import {toast} from "react-toastify";


class BoardComponent extends Component {
    constructor(props) {
        super(props);

        this.state = {
            refreshBoard: this.props.refreshBoard,
            currentField: null,
            targetField: null,
            gamePicture: this.props.gamePicture
        };
    }


    //passing data between components
    currentFieldCallbackBoard = (dataFromChild) => {
        this.setState({currentField: dataFromChild});
    };

    //passing data between components
    targetFieldCallbackBoard = async (dataFromChild) => {
        if (this.state.currentField != null) {
            await this.setState({targetField: dataFromChild});

            await this.doMove();

            await this.props.refreshBoard();

        }
    };

    doMove = async () => {

        let currentField = this.state.currentField;
        let targetField = this.state.targetField;
        let requestObject = {
            currentField,
            targetField
        };

        const {data} = await GameDataService.doMove(requestObject);

        await this.setState({currentField: null});
        await this.setState({targetField: null});

        if(!data.isValid) toast.error(data.validationMessage);

        return data;
    };

    render() {

        return (
            <div>
                {
                    this.props.gamePicture.map(fieldRow => (
                        <FieldLineComponent key={fieldRow.id} fieldRow={fieldRow.fields}
                                            currentFieldCallbackBoard={this.currentFieldCallbackBoard}
                                            targetFieldCallbackBoard={this.targetFieldCallbackBoard}
                                            moveHistory={this.props.moveHistory}
                        />
                    ))
                }
            </div>
        )
    }
}

export default BoardComponent
