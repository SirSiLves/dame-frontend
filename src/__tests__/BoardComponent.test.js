import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import BoardComponent from '../component/BoardComponent';

it('renders Board without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BoardComponent gamePicture={gamePicture} />, div);
});

describe('Board Component', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<BoardComponent gamePicture={gamePicture} debug />);

        expect(component).toMatchSnapshot();
    });
});

const gamePicture = [
    {
        id: 0,
        fields: [
            {
                fieldColor: "SANDYBROWN",
                fieldNumber: 0,
                stone: null
            },
            {
                fieldColor: "BROWN",
                fieldNumber: 1,
                stone: {stoneId: 0, stoneColor: "BLACK", queen: false, alive: true}
            }
        ]
    },
    {
        id: 1,
        fields: [
            {
                fieldColor: "BROWN",
                fieldNumber: 8,
                stone: {stoneId: 4, stoneColor: "BLACK", queen: false, alive: true}
            },
            {
                fieldColor: "SANDYBROWN",
                fieldNumber: 9,
                stone: null
            }
        ]
    }
    ];