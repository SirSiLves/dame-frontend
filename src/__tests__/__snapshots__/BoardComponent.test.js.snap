// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
